### PR TITLE
[HOTFIX] Super catchup plus restart libp2p plus upload blocks

### PR DESCRIPTION
**Pull request by es92**

### CODE REVIEW

**Comment by bkase** [comment]
1da42ec, 203e664, 5548ab6 are reviewed as part of #7363

**Comment by bkase** [comment]
0a8fdee is reviewed as part of #7244

**Comment by bkase** [comment]
1e77355 is okay for now, @mrmr1993 fixed this properly and it will eventually become a part of this PR

**Comment by bkase** [comment]
d798e20 is actually an indication that we implemented precomputed block logging incorrectly -- and indeed we had (the default was incorrectly set to true). I will revert this commit since it should be fixed for now.

File: src/lib/transition_router/transition_router.ml [hunk 1]
```diff
@@ -188,31 +188,33 @@ let download_best_tip ~logger ~network ~verifier ~trust_system
...
-                |> Length.to_int
-              in
-              Mina_metrics.Transition_frontier.update_max_blocklength_observed
-                best_tip_length ;
-              don't_wait_for
-              @@ Broadcast_pipe.Writer.write most_recent_valid_block_writer
-                   candidate_best_tip.data ;
-              enveloped_candidate_best_tip )
-            else enveloped_existing_best_tip ) )
+            let f x =
```
Comment (bkase) [inline src/lib/transition_router/transition_router.ml:192]: this change is a nice bugfix and it is fine that it's here ✅ \

File: src/lib/ledger_catchup/super_catchup.ml [hunk 2]
```diff
@@ -492,28 +497,25 @@ module Downloader = struct
...
   [%test_eq: int]
     (Downloader.total_jobs downloader)
     (Hashtbl.count t.nodes ~f:(fun node ->
          Node.State.enum node.state = To_download ))
 
+let download s d ~key ~attempts =
+  let logger = Logger.create () in
+  [%log debug]
+    ~metadata:[("key", Downloader.Key.to_yojson key); ("caller", `String s)]
+    "Mownload download $key" ;
```
Comment (mrmr1993) [inline src/lib/ledger_catchup/super_catchup.ml:516]: Nit: `s/Mownload/Download`

**Review by mrmr1993 — APPROVED** [review]

**Comment by bkase** [comment]
Please treat this as a HOTFIX branch

**Comment by bkase** [comment]
Abandoning, and will reopen in a moment

**Pull request closed without merging**